test/e2e/network/service.go

```diff
@@ -3282,42 +3280,60 @@ var _ = SIGDescribe("ESIPP [Slow]", func() {
 			framework.Failf("Service HealthCheck NodePort still present")
 		}
 
-		endpointNodeMap, err := jig.GetEndpointNodes()
+		config := e2enetwork.NewNetworkingTestConfig(f, false, false)
+
+		epNodes, err := jig.GetEndpointNodeNames()
 		framework.ExpectNoError(err)
+		endpointNodeMap := make(map[string][]string)
 		noEndpointNodeMap := map[string][]string{}
 		for _, n := range nodes.Items {
-			if _, ok := endpointNodeMap[n.Name]; ok {
-				continue
+			ips := e2enode.GetAddresses(&n, v1.NodeInternalIP)
+			if len(ips) == 0 {
+				framework.Failf("No internal ip found for node %s", n.Name)
+			}
+			if epNodes.Has(n.Name) {
+				endpointNodeMap[n.Name] = ips
+			} else {
+				noEndpointNodeMap[n.Name] = ips
 			}
-			noEndpointNodeMap[n.Name] = e2enode.GetAddresses(&n, v1.NodeExternalIP)
 		}
 
 		svcTCPPort := int(svc.Spec.Ports[0].Port)
 		svcNodePort := int(svc.Spec.Ports[0].NodePort)
 		ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
 		path := "/clientip"
+		dialCmd := "clientip"
 
 		ginkgo.By(fmt.Sprintf("endpoints present on nodes %v, absent on nodes %v", endpointNodeMap, noEndpointNodeMap))
 		for nodeName, nodeIPs := range noEndpointNodeMap {
-			ginkgo.By(fmt.Sprintf("Checking %v (%v:%v%v) proxies to endpoints on another node", nodeName, nodeIPs[0], svcNodePort, path))
-			GetHTTPContent(nodeIPs[0], svcNodePort, e2eservice.KubeProxyLagTimeout, path)
+			ginkgo.By(fmt.Sprintf("Checking %v (%v:%v/%v) proxies to endpoints on another node", nodeName, nodeIPs[0], svcNodePort, dialCmd))
+			GetHTTPContentFromTestContainer(config, nodeIPs[0], svcNodePort, e2eservice.KubeProxyLagTimeout, dialCmd)
 		}
 
 		for nodeName, nodeIPs := range endpointNodeMap {
 			ginkgo.By(fmt.Sprintf("checking kube-proxy health check fails on node with endpoint (%s), public IP %s", nodeName, nodeIPs[0]))
-			var body bytes.Buffer
+			var body string
 			pollfn := func() (bool, error) {
```

Comment:
This retry logic looks awfully similar to the `GetHTTPContentFromTestContainer` logic. Cannot you just call `GetHTTPContentFromTestContainer` here?

Reply:
I guess we cannot just call `GetHTTPContentFromTestContainer` here, since the retry logic here is slightly different from `GetHTTPContentFromTestContainer`. It expects error here while `GetHTTPContentFromTestContainer` expects success response.

Reply:
I see, makes sense, thanks for explaining